scripts/apps/workspace/content/controllers/ContentProfileFields.ts

```diff
@@ -118,7 +118,7 @@ export default function ContentProfileFields($scope, content, vocabularies, meta
                 this.sections[section].enabled.push(key);
                 this.model.editor[key].order = index + 1; // keep order in sync
                 // keep required value of editor fields in sync with the schema fields SDNTB-615
-                this.model.editor[key].required = this.model.schema[key].required;
+                this.model.editor[key].required = this.model?.schema[key]?.required ?? false;
```

Comment:
I see `this.model.schema` used to be accessed unconditionally. Can `this.model` be null? Optional chaining is a great feature, but we should not overuse it.

Reply:
I'm not sure about `this.model.schema` but `key` is definitely missing in some cases.

Reply:
Is there a reason to be missing? It doesn't look like it to me. Can you debug and see if you can find why it's happening?

Reply:
Ok, so the issue is with the content profiles that are prepopulated, for ex: belga_text CP has 38 fields in the `editor` and 29 fields in the `schema` field. Although no such case with the ones created from the UI.

Reply:
> belga_text CP has 38 fields in the editor and 29 fields in the schema field.

Can you investigate it further to see what's the difference between `schema` and `editor` and whether we should support having a different number of fields inside them?

With my limited understanding, it looks like the fields in `schema` and `editor` should be identical. If that ends up being the case, update the prepopulation data and throw an error on the client if the key is missing.

Reply:
There are certain CVs that will look like a separate field in the editor but their value will be stored inside the different fields (ex:`news_service` inside `subject`).  So in the content profile, `news_service` will be declared as a separate field inside the editor, and will be declared inside `subject` in case of `schema`. So, we will have a different number of fields inside the editor and schema.

![belga_text](https://user-images.githubusercontent.com/10600741/69625263-3b8f9780-106c-11ea-8803-ead3b32f9bde.png)

Reply:
I see. Let's keep the key optional then and add a comment explaining it in the code.